src/validator/MatchValidator.cpp

```diff
@@ -755,6 +750,103 @@ Status MatchValidator::validateYield(YieldClauseContext &yieldCtx) const {
     }
 }
 
+Status MatchValidator::isAliasDefined(const std::unordered_map<std::string, AliasType> *aliasesUsed,
+                                      const std::string *name) const {
+    if (!aliasesUsed || aliasesUsed->count(*name) != 1) {
+        return Status::SemanticError("Alias used but not defined: `%s'", name->c_str());
+    }
+    return Status::OK();
+}
+
+StatusOr<AliasType> MatchValidator::getAliasType(
+    const std::unordered_map<std::string, AliasType> *aliasesUsed,
+    const std::string *name) const {
+    NG_RETURN_IF_ERROR(isAliasDefined(aliasesUsed, name));
+    auto iter = aliasesUsed->find(*name);
+    return iter->second;
```

Comment:
Could the `iter` not be the `aliasesUsed.end()` ? If so, should `DCHECK` it.

Reply:
`isAliasDefined()` has ensured the presence of `name`, so `iter` will always point to a valid position.